### ISSUE TITLE
Metric L1 fails for multi-class data

### ISSUE THREAD

**Issue by mqlenhamey** [w4k2/problexity#1]
The attached Jupyter notebook demonstrates that problexity fails to perform metric L1 on multi-class data.  This is the only metric that fails.  Lerona(2019) defines binary metrics on multi-class data using the one-vs-one approach.  It is strange that problexity works for L2 and L3 but not L1 on multi-class data.

[px-1-fails-l1-multi-class - Jupyter Notebook.pdf](https://github.com/w4k2/problexity/files/9613020/px-1-fails-l1-multi-class.-.Jupyter.Notebook.pdf)

Here is the Python code extracted from Jupyter notebook (since I cannot attach it as a text file)

#!/usr/bin/env python
# coding: utf-8

# In[4]:

# The following Python code demonstrates that problexity fails to compute 
# l1 on multi-class data.  l1 is the only metric that fails.
# The original paper (Lorena 2019) defines all the metrics including l1 
# for multi-class data by one-versus-one calculation and averaging of the metrics.

# The numpy version is 1.22.3.
# The problexity version is 0.3.4.
# The scikit-learn version is 1.0.2.

import problexity as px
import numpy as np

# Loading binary benchmark dataset from scikit-learn
from sklearn.datasets import load_breast_cancer
X, y = load_breast_cancer(return_X_y=True)

# Set up the default configuration with all metrics for binary task.
# Initialize ComplexityCalculator with default parametrization
cc_binary = px.ComplexityCalculator()

# Fit model with data. This will fail with multi-class data.
print ("Binary with l1:")
cc_binary.fit(X,y)
print (cc_binary.report()['complexities'])

# Loading multi-class benchmark dataset from scikit-learn
from sklearn.datasets import load_iris 
X, y = load_iris(return_X_y=True)

# First we demonstrate that every metric except L1 works with the multi-class data iris
# Code adapted from https://github.com/w4k2/problexity/blob/master/problexity/ComplexityCalculator.py
from problexity import f1, f1v, f2, f3, f4
from problexity import l1, l2, l3
from problexity import n1, n2, n3, n4, t1, lsc
from problexity import density, clsCoef, hubs
from problexity import t2, t3, t4
from problexity import c1, c2

my_METRICS = [f1, f1v, f2, f3, f4, l1, l2, l3, n1, n2, n3, n4,
             t1, lsc, density, clsCoef, hubs, t2, t3, t4, c1, c2]
my_COLORS = ['#FD0100', '#F76915', '#EEDE04', '#A0D636', '#2FA236', '#333ED4']
my_RANGES = {'FB': 5, 'LR': 3, 'NB': 6,
            'NE': 3, 'DM': 3, 'CI': 2}
my_RANGES['FB'] = 5
my_RANGES['LR'] = 2  # Metric l1 fails for multi-class data - exclude it
my_RANGES['NB'] = 6
my_RANGES['NE'] = 3
my_RANGES['DM'] = 3
my_RANGES['CI'] = 2
# Exclude l1 fom the list of metrics:
cc_multi = px.ComplexityCalculator(metrics=my_METRICS[0:5]+my_METRICS[6:22], colors=my_COLORS, ranges=my_RANGES.copy())

# Fit model with data and print the report. This will succeed with multi-class data.
cc_multi.fit(X,y)
print ("Multi-class without l1:")
print (cc_multi.report()['complexities'])

# Now we use the default configuration with all metrics.
# Initialize ComplexityCalculator with default parametrization
cc = px.ComplexityCalculator()

# Fit model with data. This will fail with multi-class data.
print ("Multi-class with l1:")
cc.fit(X,y)
print (cc.report()['complexities'])
